### PR TITLE
 Add new BenderBox application (with 3 tools)

### PR DESCRIPTION
**Pull request by VanyaBelyaev**
Add new BenderBox application with three tools: 

- BenderModule
- bender
- ostap

### CODE REVIEW

**Comment by alexanderrichards** [comment]
test this please

File: python/GangaLHCb/Lib/Applications/BenderBox.py [hunk 1]
```diff
@@ -0,0 +1,521 @@
+# -*- coding: utf-8 -*-
```
Comment (egede) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: Does this line cause trouble? No other file in the Ganga release tree has this.

File: python/GangaLHCb/Lib/Applications/BenderBox.py [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+#  - number of event to process
+#
+#  Usage:
+#  @code 
+#  j.application = BenderBox ( tool = BenderModule ( module = 'the_path/the_module.py' , events = 1000 ) )
+#  @endcode 
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+class BenderModule(GangaObject):
+    """Helper class to define the main proeprties of Bender applicatiom
```
Comment (egede) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: proeprties -> properties

File: python/GangaLHCb/Lib/Applications/BenderBox.py [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+    - the name of Bender module to run
+    - dictionary of parameters to be forwarder to <code>configure</code> method
+    - number of event to process
+
+    ======
+    Usage:
+    ======
+    
+    j.application = Bender ( component = BenderModule ( module = 'the_path/the_module.py' , events = 1000 ) )
```
Comment (egede) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: Should be BenderBox here I guess. This comment applies to many places.

File: python/GangaLHCb/Lib/Applications/BenderBox.py [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+#  > bender [ scripts [ scripts ...  --no-color [ arguments ] --import [ imports [ imports [ ... --no-castor --import=data.py --batch   [ --command  [ commands [ ... 
+#  @endcode
+#  Usage:
+#  @code 
+#  j.application = BenderBox ( tool = bender ( scripts   = ['path_to_script/the_script.py']  ,
+#                                              commands  = [ 'print ls()' ]  ) )
+#  @endcode 
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+class bender(GangaObject):
```
Comment (egede) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: There should be no lowercase class names. This is in particular confusing with the `Bender` GPI name of the legacy `Bender` application type.

File: python/GangaLHCb/Lib/Applications/BenderBox.py [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+#  @endcode
+#  Usage:
+#  @code
+#  j.application = BenderBox ( tool = Ostap( scripts   = ['path_to_script/the_script.py']  ,
+#                                            arguments = [ '--no-canvas' ]  ,
+#                                            commands  = [ 'print dir()' ]  ) ) 
+#  @encode
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+class ostap(GangaObject):
```
Comment (egede) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: Again no lowercse.

File: python/GangaLHCb/Lib/Applications/BenderBox.py [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+            arguments = self.arguments  ,
+            command   = self.commands    
+            )
+        return the_script
+
+## import the base class 
+from .GaudiExec import GaudiExec
+# =============================================================================
+## @class BenderBox
+#  Box with Bender-based applications
```
Comment (egede) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: Can we maybe add a URL to where there is a description of the Bender applications in general?

File: python/GangaLHCb/Lib/Applications/BenderBox.py (hunk 1 above)
Comment (VanyaBelyaev) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: it is highly useful, and e,g it was just vital at SVN epoche. when SVN automatically sustituted some pattersm e.g. authors with not-ASCII symbols (e.g. French names)...  and it caused problem for subsequent python.zip.   Finally I've got a habit to always use this declaration in python code

File: python/GangaLHCb/Lib/Applications/BenderBox.py (hunk 6 above)
Comment (VanyaBelyaev) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: Not sure that it is needed. Only people who know what is Bender project will use BenderBox in Ganga. For others is has no interest..

File: python/GangaLHCb/Lib/Applications/BenderBox.py (hunk 4 above)
Comment (VanyaBelyaev) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: lowercase name matches well with exact bender-script, as people use them interactively.

File: python/GangaLHCb/Lib/Applications/BenderBox.py (hunk 5 above)
Comment (VanyaBelyaev) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: lowercase name matches well with exact ostap-script, as people use them interactively.

File: python/GangaLHCb/Lib/Applications/BenderBox.py (hunk 3 above)
Comment (VanyaBelyaev) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: will fix it as soon as manage how to push... I have some problems..

File: python/GangaLHCb/Lib/Applications/BenderBox.py (hunk 2 above)
Comment (VanyaBelyaev) [inline python/GangaLHCb/Lib/Applications/BenderBox.py]: easy to fix...

**Comment by mesmith75** [comment]
Can we remove this one?

**Comment by VanyaBelyaev** [comment]
@mesmith75 yes, please

**Pull request closed without merging**